contrib/storage-mongo/src/test/java/org/apache/drill/exec/store/mongo/MongoTestSuit.java

```diff
@@ -234,15 +240,17 @@ private static void createDbAndCollections(String dbName,
 
   @AfterClass
   public static void tearDownCluster() throws Exception {
-    if (mongoClient != null) {
-      mongoClient.dropDatabase(EMPLOYEE_DB);
-      mongoClient.close();
-    }
     synchronized (MongoTestSuit.class) {
-      if (distMode) {
-        DistributedMode.cleanup();
-      } else {
-        SingleMode.cleanup();
+      if (initCount.decrementAndGet() == 0) {
+        if (mongoClient != null) {
+          mongoClient.dropDatabase(EMPLOYEE_DB);
+          mongoClient.close();
+        }
+        if (distMode) {
+          DistributedMode.cleanup();
+        } else {
+          SingleMode.cleanup();
```

Comment:
Should the above include error handling? It seems that if the `dropDatabase` fails, then `close()` won't be called, nor will the `cleanup` method.